src/RadioGroup/RadioGroup.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC } from 'react'
+import { useUniqueId } from '../utils/id'
+import { RadioItem } from './RadioItem'
+import styled from 'styled-components'
+
+export type RadioGroupProps = {
+  options: Array<{
+    label: string
+    value: string
+  }>
+  onChange: (value: string) => void
+  value: string
+}
+
+export const RadioGroup: FC<RadioGroupProps> = ({
+  options,
+  onChange,
+  value,
+}) => {
+  const name = useUniqueId()
```

Comment:
Why is a unique id set as the name here then passed down to the `RadioItem` and `RadioElement` - how is the name used?

Reply:
`useUniqueId` is an utility to generate an unique identifier, it's not related to the `id` prop per say even if it's the most common usage

Here we generate an unique identifier for the `name` prop on each option so that they are linked together